### PR TITLE
[TD-12102]<fix>:csharp support fetch_lengths,fix fetch_row binray ret…

### PR DESCRIPTION
**Pull request by xleili**
…rieve error

### CODE REVIEW

File: src/connector/C#/src/TDengineDriver/TDengineDriver.cs [hunk 1]
```diff
@@ -395,5 +395,8 @@ static public string StmtErrorStr(IntPtr stmt)
             IntPtr stmtErrPrt = StmtErrPtr(stmt);
             return Marshal.PtrToStringAnsi(stmtErrPrt);
         }
+
+        [DllImport("taos", EntryPoint = "taos_fetch_lengths", CallingConvention = CallingConvention.Cdecl)]
+        static extern public IntPtr FetchLength(IntPtr taos);
```
Comment (zitsen) [inline src/connector/C#/src/TDengineDriver/TDengineDriver.cs]: Make the name comparable. FetchLength -> FetchLengths.

File: src/connector/C#/src/test/Cases/FetchLength.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+            UtilsTools.ExecuteQuery(conn,sql1);
+            UtilsTools.ExecuteQuery(conn,sql2);
+            UtilsTools.ExecuteQuery(conn,sql3);
+            UtilsTools.ExecuteQuery(conn,sql4);
+            UtilsTools.ExecuteQuery(conn,sql5);
+
+            IntPtr res = IntPtr.Zero;
+            res = UtilsTools.ExecuteQuery(conn,sql6);
+            UtilsTools.DisplayRes(res);
```
Comment (zitsen) [inline src/connector/C#/src/test/Cases/FetchLength.cs]: Compare the data to expected in test case, not only display.

**Review by sangshuduo — APPROVED** [review]
LGTM

**Review by zitsen — CHANGES_REQUESTED** [review]
Please update.

File: src/connector/C#/src/test/Cases/FetchLength.cs (hunk 2 above)
Comment (xleili) [reply src/connector/C#/src/test/Cases/FetchLength.cs]: I'will fix, and update this test case later.

File: src/connector/C#/src/TDengineDriver/TDengineDriver.cs (hunk 1 above)
Comment (xleili) [reply src/connector/C#/src/TDengineDriver/TDengineDriver.cs]: ok, i will fix it

Comment (xleili) [reply src/connector/C#/src/TDengineDriver/TDengineDriver.cs]: done

File: src/connector/C#/src/test/Cases/FetchLength.cs (hunk 2 above)
Comment (xleili) [reply src/connector/C#/src/test/Cases/FetchLength.cs]: done

**Review by zitsen — APPROVED** [review]
LGTM

**Pull request merged**